File: backend-node-portifolio/src/api/domain/models/Like.ts
import { Comment } from "./Comment";
import { Post } from "./Post";
import { User } from "./User";

export class Like{
    public Id: number;
    public Guid: string;
    public UserId: number;
    public User: User;
    public PostId: number;
    public Post: Post;
    public CommentId: number;
    public Comment: Comment;
    constructor() {
        this.Id = 0;
        this.Guid = "";
        this.UserId = 0;
        this.User = new User();
        this.PostId = 0;
        this.Post = new Post();
        this.CommentId = 0;
        this.Comment = new Comment()
    }
}

File: backend/PostService/src/PostService.Domain/Entities/Like.cs
namespace PostService.Domain.Entities
{
    public class Like
    {
        public Guid Id { get; private set; }
        public Guid PostId { get; private set;}
        public Post Post { get; private set; }
        public Guid UserId { get; private set; }
        public DateTime CreatedAt { get; private set; }
        public Like(Guid userId, Guid postId)
        {
            this.UserId = userId;
            this.PostId = postId;
            this.CreatedAt = DateTime.UtcNow;
        }
    }
}